packages/codegen-ui-react/lib/react-studio-template-renderer.ts

```diff
@@ -594,7 +594,7 @@ export abstract class ReactStudioTemplateRenderer extends StudioTemplateRenderer
       factory.createBindingElement(
         factory.createToken(ts.SyntaxKind.DotDotDotToken),
         undefined,
-        factory.createIdentifier('rest'),
+        factory.createIdentifier(hasVariant ? 'restProp' : 'rest'),
```

Comment:
What is the purpose of using 'restProp' vs 'rest'?

Reply:
If a component has variants, we want to include a default transition style on the parent component. In a component-with-variants case, using restProp initially lets us define rest with a default transition style later (see line 778 in react-studio-template-renderer.ts)